### PR TITLE
Coding standards tweaks to daskified cube code

### PR DESCRIPTION
**Pull request by dkillick**
There were a few inconsistencies in the `cube` + `dask` code, and instead of noting them down for further discussion, I thought I'd make a vessel for discussion (that is, this PR).

A retargetting of #2415 to the base `dask_timed`.

### CODE REVIEW

**Comment by dkillick** [comment]
See the comments on #2415 for a discussion of why I've made these changes.

**Comment by dkillick** [comment]
Ping @bjlittle @pp-mo @lbdreyer

**Comment by pp-mo** [comment]
> See the comments on #2415 for a discussion of why I've made these changes.

See related query, here : https://github.com/SciTools/iris/pull/2415/files#r104770825

File: lib/iris/cube.py [hunk 1]
```diff
@@ -2205,12 +2205,11 @@ def new_cell_measure_dims(cm_):
...
             first_slice = None
-        if self._numpy_array is not None:
-            cube_data = self._numpy_array
-        elif self._dask_array is not None:
+
+        if self.has_lazy_data():
             cube_data = self._dask_array
         else:
-            raise ValueError('This cube has no data, slicing is not supported')
+            cube_data = self._numpy_array
```
Comment (bjlittle) [inline lib/iris/cube.py]: @dkillick At this point, we're confident that either `self._dask_array` or `self._numpy_array` is populated.

Prior to this change, there seems to be a hint of that might not be the case, hence the `ValueError` exception being raised. I'd certainly hope that there is a data payload of some kind to process - we need to make sure that internally that is always the case.

**Comment by bjlittle** [comment]
@dkillick @pp-mo @lbdreyer I'm :+1: on this PR ... are there any reasons why we can't bank this now?

File: lib/iris/cube.py (hunk 1 above)
Comment (dkillick) [inline lib/iris/cube.py]: I think if we end up at this point having a cube with no data we're deeper in the sticky than a `ValueError` will help with! The rest of the code in the class makes it clear that we can never _construct_ a cube with no data, so if an end user has done something that means their cube no longer has any data then IMO they can dig their own way out of the sticky stuff...

**Comment by dkillick** [comment]
@bjlittle let's just make sure @pp-mo is happy with my comment at his link above :smile:

**Comment by dkillick** [comment]
@bjlittle I think @pp-mo and I might be happy with this now - we invite you to hit the merge button once Travis is happy!

File: lib/iris/cube.py (hunk 1 above)
Comment (bjlittle) [inline lib/iris/cube.py]: I'm really hoping that there never, ever is any `sticky stuff` :wink:

It's just a note to ourselves really, that we need to ensure the integrity of the cube as best we can

**Pull request merged**